server/webhook.go

```diff
@@ -150,8 +145,19 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, err).SetInternal(err)
 		}
 
-		//nolint:misspell
-		// TODO(unknwon): Validate webhook signature, https://linear.app/bbteam/issue/BYT-935
+		body, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to read webhook request").SetInternal(err)
+		}
+
+		if !validateGitHubWebhookSignature256(c.Request().Header.Get("X-Hub-Signature-256"), repo.WebhookSecretToken, body) {
+			return echo.NewHTTPError(http.StatusBadRequest, "Mismatched payload signature")
+		}
+
+		var pushEvent github.WebhookPushEvent
+		if err := json.Unmarshal(body, &pushEvent); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed push event").SetInternal(err)
+		}
```

Comment:
There is no point in unmarshalling the request body if the signature doesn't match, thus putting it after the signature validation.